src/guild_configs/lbp.js

```diff
@@ -0,0 +1,914 @@
+/// <reference path="../types.d.ts" />
+
+import Category from "../Category.js";
+import { MULTI_GAME } from "../misc.js";
+import * as misc from "../misc.js";
+import Race from "../Race.js";
+
+import Discord from "discord.js";
+import https from "https";
+
+const emotes = {
+	ppjE: "<:ppjE:230442929859198977>",
+	ppjSmug: "<:ppjSmug:230442929838227457>",
+	ppjWink: "<:ppjWink:367254404811980801>",
+	rbdBingo: "<:rbdBingo:394255134340677634>"
+};
+
+const ilAliases = [ "il", "ils" ];
+const anyNCAliases = [ "anynooverlord", "nocreate", "nooverlord", "anync", "anyno", "nc", "no" ];
+
+const ilCategory = {
+	aliases: ilAliases,
+	il: true
+};
+
+const COOP_REGEX = /coop|[2-5]p(layers?)?/;
+const LIGHTHOUSE_REGEX = /[a-z\.]*\.[a-z]{2,4}\/slot\/[1-9]*/mi;
+
+/**
+ * @param {string} input
+ * @returns {{ name: string; coop: boolean; }}
+ */
+function lbpCleanUpCategory(input) {
+	const name = input.toLowerCase().replace(/\W/g, "");
+	return {
+		name: name.replace(COOP_REGEX, ""),
+		coop: COOP_REGEX.test(name)
+	};
+}
+
+/**
+ * @param {string} input
+ * @returns {string}
+ */
+function lbpCleanUpLevelName(input) {
+	return input.toLowerCase()
+		.replace(/&/g, "and")
+		.replace(/\W|^the/g, "")
+		.replace("introduction", "intro");
+}
+
+/**
+ * @param {Discord.Message} message
+ * @param {Discord.GuildMember} member
+ * @param {string} args
+ * @param {string} cleanArgs
+ * @returns {boolean}
+ */
+function lbpCommunityLevels(message, member, args, cleanArgs) {
+	var match = message.content.match(LIGHTHOUSE_REGEX);
+	if (match = null) {
+		return false;
+	}
+	var level = `https://${match[0].trim()}`;
+	chooseCommunityLevel(level, message);
+	return true;
+}
+function chooseCommunityLevel(level, message) {
+    "use-strict";
+    https.get(level, (result) => {
+        var { statusCode } = result;
+        if (statusCode === 302) {
+            chooseCommunityLevel(result.headers.location, message, onEnd);
+            return;
+        }
+        if (statusCode !== 200) {
+            message.channel.send(`Couldn't follow ${level}; got a ${statusCode} response.`);
+            return;
+        }
+
+        var dataQueue = "";
+        result.on("data", (dataBuffer) => {
+            dataQueue += dataBuffer;
+        });
+        result.on("end", () => {
+            var start = dataQueue.search(`<h1>`) + 4;
+            var end = dataQueue.search(`</h1>`);
+            var title = misc.decodeHTML(dataQueue.substring(start, end).trim());
+
+			const { race } = message.channel;
+			race.level = title;
+            message.channel.send(`Level updated to ${title}.`);
```

Comment:
It would be nice if you'd deduplicate this by having `config.race.communityLevels` return the level name or `null` (instead of a bool indicating success) because `command_modules/race_control.js` should do this part instead. (The "official level" check done there should be skipped for community levels, everyone knows that community levels are not official.)